packages/library-base/src/commonMain/kotlin/io/realm/kotlin/ext/RealmListExt.kt

```diff
@@ -42,3 +47,22 @@ public inline fun <reified T : TypedRealmObject> RealmList<T>.copyFromRealm(dept
     return this.getRealm<TypedRealm>()?.copyFromRealm(this, depth)
         ?: throw IllegalArgumentException("This RealmList is unmanaged. Only managed lists can be copied.")
 }
+
+/**
+ * Query the objects of a list by the `filter` and `arguments`.
+ *
+ * @param filter the Realm Query Language predicate to append.
+ * @param arguments Realm values for the predicate.
+ */
+// Added as an extension method as we cannot add the method `fun query(...): RealmQuery<T>` to the
+// `RealmList` interface as `RealmQuery` has an `BaseRealmObject` upper bound which `RealmList` do
+// not.
+public fun <T : BaseRealmObject> RealmList<T>.query(
+    filter: String = TRUE_PREDICATE,
```

Comment:
It is called `query` in our other API's

Reply:
Seem like we use them both somehow spread across the SDK 🤪 ... maybe `predicate` that would also align well with 
```
const val ALWAYS_TRUE: String = "TRUE_PREDICATE"
```
?

Reply:
Agreed that we would keep `filter` as we already use that in various places and `query` is annoying as it is shadowing the method name.